mach-eight/reup.romulo
# Release Oct 22 2024 (#249)

* set ASTC compression format when building

* Fix tests

* fix if order

* refactor change material

* Comments addressed

* fix indentation

* Fix asset

* move ObjectVisibilityUtils to editor folder

* organize Tests to property have edit mode tests

* fix assembly definition

* separate mocks

* Change milimeters for millimeters

* update to version 2.0.2

* Fixing test

* Add check for disable objects at build time

* Add error message

* Round super small numbers to zero in matrix

* Test rounding to zero

* use ternary operator

* Fix ternarian

* update to version 2.0.3

* Fix issue where objects remain highlighted after being selected on mobile

* create AppProxy shell

* create first test of ViewModeManager

* Convert manager to a controller

* Fix namespace

* Activate FPV mediator test

* first test with character position manager spy

* install 2 cinemachine cameras

* send actions from app proxy

* Add cinemachine brain to camera

* Change cameras active status

* clean up

* Deactivate character when in DHV

* create activate all objects funcionality

* move dhv camera in horizontal plane

* Add test for passing dependency to dhv behavior

* Create ViewModeManagerTest

* Test ASDW keys to move character

* Set Input in reup scene instantiator

* disable dhv camera to start

* Test moving with the mouse

* Test tap movement

* test rotation with touch

* Test horizontal rotation with mouse

* test vertical rotation using pointers

* test horizontal rotation with arrow keys

* Test change height and initial height spawn

* test non collinearity in uvs as well

* Comment addressed

* Fix triangle ordering

* Implement new action map

* Fix disable input

* Change the CompareTag for a direct string comparison, avoiding problems if the tag is not set in the project because no blue bubbles or space selectors were used.

* Check for multiple cameras at build time

* Refactor

* Create asdw tests for move dhv camera

* Test deactivate dhv wrapper

* Fix drag manager outside of character

* Create MovePointerUtils

* Add first test to move with mouse

* create test to avoid moving dhv camera when not dragging

* Add tests to move dhv camera in touch

* Fix tests

* Remove over used prefabs and ScripHolders directory and fix tests

* Fix tests

* Improve warning message

* place camera at 50 meters from main dhv wrapper

* Restrict camera movement and fix tests

* Clean unused dependencies

* Comments addressed

* Comments addressed

* Update buildVersion to "2024-09-24" and package version to "2.0.4"

* Create TexturesManager

* Destroy a texture

* implement records

* add extra test to manager

* support protected textures

* Fix rest of the tests

* Remove event system references

* modify spaces record

* Drop class ModelInfoMessage and use JObject instead

* Refactor tests

* Zoom Dhv Camera

* Update MoveDhvCamera with serialized fields

* create first test to receive space selector list

* Send jump point's names

* return ids in space selector list

* Fix tag filters being reset on tag addition

* create first test for SpacesRecord

* move to space point

* Hide loading spinner when changing a material fails

* Create a prefab to limit player movement in external view

* Receive message in edit mediator

* Send success message

* remove dead code and assets

* Test high level feature

* Clean non-necessary

* use a Dictionary

* Refactor container

* refactor container

* fix registry

* update version date

* Comments Addressed

* fix editor tool outside the editor forlder

* DragManager Interface

* catch white space spaceJump ids

* fix tests

* Hide and show objects implementation

* Update Tests/PlayMode/TexturesManagerTest.cs



* modify edit mediator dependency injector

* resolve comments

* Comments addressed

* send not found message

* send error message if space point has no ground below

* send fail messages

* fix tests

* refactor test to be more high level

* return success messages

* fix test variables

* send interrupted message

* refactor schema

* add regex to search tag section

* deactivate filter feature

* exclusive/inclusive filters

* fix erroMessage property

* Clamp scroll input to avoid high sensitivity on trackpad

* generalize activate view mode message

* Fix appProxy tool

* Fix tests

* Add failure tests and update enum names

* Comments Addressed

* update dollhouse enum name

* escape errors in regex

* register objects in Awake

* Update package version to "2.0.5" and buildVersion to "2024-10-09"

* Improve character navigation and colliders

* Fix textures being destroyed

Textures were being destroyied when applying a texture to an object that already had that texture, and it was the only object with such texture

* add show whole building button to tag scanner tool

* delete postprocessing asset

* allow look for regex in substring filter

* deactivate house container when entering dhv

* Fix Reup prefab for dhv behaviour

* start with 40 degree angle in dhv camera

* apply antialiasing in main camera

* Fix tests

* Test movement

* remove unnused inputs

* Add zenject package to romulo

* create not passing test

* use ray utils to generate ray

* inject character position manager with zenject

* inject character position manager with zenject

* Fix test

* clean comments

* refactor instantiator

* pass first test

* failure test passing

* Improve DHV zoom tests

* apply building layer to all gameobjects inside building

* Update Editor/TagScannerTool.cs



* remove dead variables

* Destroy used GameObjects in Unity tests

* clean comments

* Delete transformHandleObj

* Fix texturesManager

* Sync objectIdsToTexturesRecord and texturesToObjectIdsRecord when a protected/default material is applied

* Add gizmo

* update URP_asset

* update version

---------
